File: DotNet/Examples/PubSubByTypeExample/PstMessage.cs
using System;
using System.Runtime.Serialization;

namespace PubSubByTypeExample
{
    [DataContract(Namespace = "")]
    class PstMessage
    {
        [DataMember] public string StringField { get; set; }
        [DataMember] public long LongField { get; set; }
        [DataMember] public DateTime DateField { get; set; }

        public override string ToString()
        {
            return string.Format("String: {0}\nLong: {1}\nDate: {2}", StringField, LongField, DateField);
        }
    }
}

File: Java/hydra-examples/src/main/java/uk/co/shastra/hydra/pubsubbytypeexample/PstMessage.java
package uk.co.shastra.hydra.pubsubbytypeexample;

import java.util.Date;

public class PstMessage {
	private String stringField;
	private long longField;
	private Date dateField;
	
	public String getStringField() { return stringField; }
	public void setStringField(String stringField) { this.stringField = stringField; }
	public long getLongField() { return longField; }
	public void setLongField(long longField) { this.longField = longField; }
	public Date getDateField() { return dateField; }
	public void setDateField(Date dateField) { this.dateField = dateField; }
	
	// Constructor for use by the deserializer, which will set properties individually.
	public PstMessage() {}
	
	public PstMessage(String stringField, long longField, Date dateField) {
		this.stringField = stringField;
		this.longField = longField;
		this.dateField = dateField;
	}
	
	@Override
	public String toString() {
		return String.format("String: %1$s\nLong: %2$s\nDate: %3$s", stringField, longField, dateField);
	}
	
}
